include/HamiltonianBuilder/FrozenCore.hpp

```diff
@@ -0,0 +1,89 @@
+// This file is part of GQCG-gqcp.
+//
+// Copyright (C) 2017-2019  the GQCG developers
+//
+// GQCG-gqcp is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// GQCG-gqcp is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-gqcp.  If not, see <http://www.gnu.org/licenses/>.
+//
+#ifndef GQCP_FROZENCOREINTERFACE_HPP
+#define GQCP_FROZENCOREINTERFACE_HPP
+
+
+#include "HamiltonianParameters/HamiltonianParameters.hpp"
+#include "HamiltonianBuilder/HamiltonianBuilder.hpp"
+
+
+namespace GQCP {
+
+
+/**
+ *  Base class containing general functions related to frozen core CI
+ */
+class FrozenCore : public HamiltonianBuilder {
```

Comment:
kept the name FrozenCore instead of FrozenCoreInterface because it isn't really an interface since it implements overriden functions from an interface